lib/galaxy/files/sources/remotezip.py

```diff
@@ -0,0 +1,194 @@
+import logging
+import zlib
+from dataclasses import (
+    dataclass,
+    fields,
+)
+from struct import unpack
+from typing import (
+    Dict,
+    Optional,
+)
+from urllib.parse import (
+    parse_qs,
+    unquote,
+    urlparse,
+)
+
+import requests
+from typing_extensions import Unpack
+
+from galaxy.files import OptionalUserContext
+from . import (
+    BaseFilesSource,
+    FilesSourceOptions,
+    FilesSourceProperties,
+    PluginKind,
+)
+
+log = logging.getLogger(__name__)
+
+
+@dataclass
+class FileExtractParameters:
+    source: str
+    header_offset: int
+    compression_method: int
+    compress_size: int
+
+
+DOC_TEMPLATE = """
+Remote ZIP archive extractor
+---------------------------
+
+This plugin extracts files from a public remotely hosted ZIP archive using byte-range requests.
+The following parameters are required and assumed to be known in advance:
+
+- `source`: The URL of the remote ZIP archive.
+- `header_offset`: The offset of the file header in the archive.
+- `compression_method`: The compression method used for the file.
+- `compress_size`: The size of the compressed file.
+"""
+
+
+class RemoteZipFilesSource(BaseFilesSource):
```

Comment:
Would this be better implemented as a mixin / extension that can be added for individual file sources ? For stuff like S3, google drive, dropbox you'll likely have proper file extensions you can use to determine if the archive can be entered ?

Reply:
I see the value of such a mixin for file sources, and I'm willing to try to implement it (I'll create a follow-up issue for it). However, this current implementation serves a slightly different purpose. This one is just for extracting and uploading single files (no exploration), for which the ZIP metadata (offset, compression size, etc.) is already known (this part is handled in the client). So it is just a "tool" to reference a file inside a ZIP archive, so it is not a typical File Source, and I see why this is confusing now... this is more in line with the base64 File source.